wifi: ap: Add support for STA disconnection

krish2718 opened the pull request.

krish2718 committed:
wifi: ap: Add support to disconnect a STA

In AP mode, this can be used to disconnect a connected station.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

krish2718 committed:
wifi: shell: ap: Add a command to disconnect a station

The shell commands can be used to disconnect a connected station in AP
mode.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

rlubos commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -1361,6 +1361,34 @@ static int cmd_wifi_ap_stations(const struct shell *sh, size_t argc,
+	uint8_t mac[6];
+	int ret;
+
+	if (argc != 2) {
+		shell_fprintf(sh, SHELL_WARNING, "Invalid number of arguments\n");
+		shell_help(sh);
+		return -ENOEXEC;
+	}
```
Isn't this check redundant with `SHELL_CMD_ARG()` usage, which handles the argument count validation on the shell side?

krish2718 replied:
Yeah, it can be removed, but having it also is okay in case we break that dependency. (True for all other commands as well).

rlubos replied:
It's not a big deal, but IMO it's just a few bytes of code wasted (+ we now have two sources of truth for argc for the command).

krish2718 replied:
I can submit a new PR that removes these arg checks and add some notes.

krish2718 replied:
They were only few, so, did this as part of this PR itself.

rlubos reviewed (commented):
That fixup commit doesn't seem correct for this PR?

krish2718 commented:
> That fixup commit doesn't seem correct for this PR?

Squashed it, my alias for squashing didn't work because of a sauce tag.

krish2718 committed:
wifi: shell: Remove duplicate argument count checks

Now that we are using the shell macro to enforce argument count check,
both mandatory and optional arguments, these additions checks are
unnecessary.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>
